wgpu-hal/src/dx12/shader_compilation.rs

```diff
@@ -0,0 +1,260 @@
+use std::ptr;
```

Comment:
I split it up in a similar way to the gpu_allocator pr, to avoid a mess of cfgs in device.rs.

Reply:
Awesome!